[Bug] Can not find type hints for Hydra

# 🐛 Bug

Running `mypy` on a hydra app throws type errors.`Skipping analyzing 'hydra': found module but no type hints or library stubs`

## To reproduce

** Minimal Code/Config snippet to reproduce **

* Create a file, say `test.py`
* Add a single line `import hydra`
* Run `mypy test.py`

** Stack trace/error message **
```
/(hydra_types) ➜  ~ mypy test.py
test.py:1: error: Skipping analyzing 'hydra': found module but no type hints or library stubs
test.py:1: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 1 source file)
```

## Expected Behavior
Type tests should pass

## System information
- **Hydra Version** :  Master and pip (1.0rc)
- **Python version** : 3.7
- **Virtual environment type and version** : Conda 4.5.8
- **Operating system** : Linux

## Additional context

https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-type-hints-for-third-party-library

--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,3 +1,5 @@
+include hydra/py.typed
 global-exclude *.pyc
 global-exclude __pycache__
 recursive-include hydra/* *
+

--- a/news/657.bugfix
+++ b/news/657.bugfix
@@ -1,0 +1,1 @@
+Add hydra/py.typed to pip package to enable type checking in downstream projects

